DFC.Digital/DFC.Digital.Services.SendGrid/SendGridEmailService.cs

```diff
@@ -34,31 +34,36 @@ public class SendGridEmailService : INoncitizenEmailService<ContactUsRequest>
             this.mapper = mapper;
         }
 
-        public async Task<bool> SendEmailAsync(ContactUsRequest sendEmailRequest)
+        public virtual async Task<bool> SendEmailAsync(ContactUsRequest sendEmailRequest)
+        {
+            var template = emailTemplateRepository.GetByTemplateName(sendEmailRequest.TemplateName);
+            return await SendEmail(sendEmailRequest, template).ConfigureAwait(false);
+        }
+
+        protected async Task<bool> SendEmail(ContactUsRequest sendEmailRequest, EmailTemplate template)
```

Comment:
Can we change this so SendEmailAsync can accept email template as optional param

```
 public virtual async Task<bool> SendEmailAsync(ContactUsRequest sendEmailRequest, EmailTemplate template = null)
        {
            if (simulateEmailResponsesService.IsThisSimulationRequest(sendEmailRequest.Email))
            {
                return simulateEmailResponsesService.SimulateEmailResponse(sendEmailRequest.Email);
            }

            if (template == null)
            {
                template = emailTemplateRepository.GetByTemplateName(sendEmailRequest.TemplateName);
            }

           
```

Reply:
Updated as suggested